pkg/tbtc/signing_test.go

```diff
@@ -202,6 +342,252 @@ func TestSigningRetryLoop(t *testing.T) {
+			message:                    big.NewInt(100),
+			broadcastingMembersIndexes: []group.MemberIndex{1, 2, 3, 4, 5},
+			expectedErrors:             make(map[group.MemberIndex]error),
+			expectedResults: map[group.MemberIndex][]group.MemberIndex{
+				1: {1, 2, 3, 4, 5},
+				2: {1, 2, 3, 4, 5},
+				3: {1, 2, 3, 4, 5},
+				4: {1, 2, 3, 4, 5},
+				5: {1, 2, 3, 4, 5},
+			},
+		},
+		"honest majority of members broadcasted announcements": {
+			message:                    big.NewInt(200),
+			broadcastingMembersIndexes: []group.MemberIndex{1, 3, 5},
+			expectedErrors:             make(map[group.MemberIndex]error),
+			expectedResults: map[group.MemberIndex][]group.MemberIndex{
+				1: {1, 3, 5},
+				3: {1, 3, 5},
+				5: {1, 3, 5},
+			},
+		},
+		"minority of members broadcasted announcements": {
+			message:                    big.NewInt(300),
+			broadcastingMembersIndexes: []group.MemberIndex{1, 3},
+			expectedErrors: map[group.MemberIndex]error{
+				1: fmt.Errorf("ready members count is lesser than the honest threshold"),
+				3: fmt.Errorf("ready members count is lesser than the honest threshold"),
+			},
+			expectedResults: make(map[group.MemberIndex][]group.MemberIndex),
+		},
+	}
+
+	for testName, test := range tests {
+		t.Run(testName, func(t *testing.T) {
+			broadcastChannel, err := localProvider.BroadcastChannelFor(
+				test.message.Text(16),
+			)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			membershipValidator := group.NewMembershipValidator(
+				&testutils.MockLogger{},
+				signingGroupOperators,
+				localChain.Signing(),
+			)
+
+			announcer := newBroadcastSigningAnnouncer(
+				chainConfig,
+				broadcastChannel,
+				membershipValidator,
+			)
+
+			resultsChan := make(
+				chan *memberResult,
+				len(test.broadcastingMembersIndexes),
+			)
+			errorsChan := make(
+				chan *memberError,
+				len(test.broadcastingMembersIndexes),
+			)
+
+			wg := sync.WaitGroup{}
+			wg.Add(len(test.broadcastingMembersIndexes))
+
+			for _, broadcastingMemberIndex := range test.broadcastingMembersIndexes {
+				go func(memberIndex group.MemberIndex) {
+					defer wg.Done()
+
+					ctx, cancelCtx := context.WithTimeout(
+						context.Background(),
+						100*time.Millisecond,
+					)
+					defer cancelCtx()
+
+					readyMembersIndexes, err := announcer.announce(
+						ctx,
+						memberIndex,
+						test.message,
+						1,
+					)
+					if err != nil {
+						errorsChan <- &memberError{memberIndex, err}
+						return
+					}
+
+					resultsChan <- &memberResult{memberIndex, readyMembersIndexes}
+				}(broadcastingMemberIndex)
+			}
+
+			wg.Wait()
+
+			close(resultsChan)
+			results := make(map[group.MemberIndex][]group.MemberIndex)
+			for r := range resultsChan {
+				results[r.memberIndex] = r.readyMembersIndexes
+			}
+
+			close(errorsChan)
+			errors := make(map[group.MemberIndex]error)
+			for e := range errorsChan {
+				errors[e.memberIndex] = e.err
+			}
+
+			if !reflect.DeepEqual(test.expectedErrors, errors) {
```

Comment:
We could also define the error as a `var` and use `testutils.AssertErrorsSame` here. I think this approach is cleaner and the linter will not complain.

Reply:
No longer relevant due to ab218401